src/main/java/com/example/springboot/dao/BookDAOService.java

```diff
@@ -0,0 +1,188 @@
+	
+	private static final String QUERY_FIND_ALL = "SELECT id, title, author, reader_id FROM `books`;";
+	private static final String QUERY_SELECT_BY_ID = "SELECT id, title, author, reader_id FROM `books` WHERE id=?;";
+	private static final String QUERY_CREATE = "INSERT INTO `books`(title, author, reader_id) VALUES(?, ?, ?);";
+	private static final String QUERY_UPDATE = "UPDATE `books` SET title=?, author=?, reader_id=? WHERE id=?;";
+	private static final String QUERY_DELETE = "DELETE FROM `books` WHERE id=?;";
+	
+	private static final String ID = "id";
+	private static final String TITLE = "title";
+	private static final String AUTHOR = "author";
+	private static final String READER_ID = "reader_id";
+	
+	/**
+	 * This method create and return a connection
+	 * @return Connection
+	 * @throws SQLException 
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection(URL, USERNAME, PASSWORD);
+	}
+	
+	/**
+	 * This method create new instance of Book and return it
+	 * @return Book
+	 * @throws SQLException 
+	 */
+	private Book getBook(ResultSet resultSet) throws SQLException {
+		return new Book(resultSet.getInt(ID), resultSet.getString(TITLE), resultSet.getString(AUTHOR), resultSet.getInt(READER_ID));
+	}
+	
+	/**
+	 * This method return all books from database
+	 * @return List<Book>
+	 */
+	@Override
+	public List<Book> findAll() {
+		List<Book> resultList = new ArrayList<>();
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL);
+			ResultSet resultSet = statement.executeQuery();
+			
+			while(resultSet.next()) {
+				resultList.add(getBook(resultSet));
+			}
+			
+			return resultList;
+			
+		} catch(SQLException | RuntimeException e) {
+			LOGGER.error(e.getMessage());
+		}
+		
+		return null;
+	}
+	
+	/**
+	 * This method looking for a book in database by given id
+	 * @param id
+	 * @return Book
+	 */
+	@Override
+	public Book selectById(Integer id) {
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_SELECT_BY_ID);
+			statement.setInt(1, id);
+			ResultSet resultSet = statement.executeQuery();
+			
+			resultSet.next();
+			Book book = getBook(resultSet);
+			
+			return book;
+			
+		} catch (SQLException | RuntimeException e) {
+			LOGGER.error(e.getMessage());
+		}
+		
+		return null;
+	}
+	
+	/**
+	 * This method create a new instance of book in database
+	 * @param book
+	 * @return Integer
+	 */
+	@Override
+	public Integer create(Book book) {
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_CREATE, Statement.RETURN_GENERATED_KEYS);
+			statement.setString(1, book.getTitle());
+			statement.setString(2, book.getAuthor());
+			statement.setInt(3, book.getReaderId());
+			statement.execute();
+			ResultSet generatedKeys = statement.getGeneratedKeys();
```

Comment:
very... very clear 3 lines of code, they are grouped together, as they do the same think, we don't want to move it to private method we will add empty space between them and don't said that here we fetch bookId. ttttsssss